docs/design/fabric-v2+/interfaces.md

```diff
@@ -304,58 +241,118 @@ public int ecall_next_block(
         [user_check] uint8_t *block_bytes, uint32_t block_size);
 ```
 
-```c++
-// this secure session functions are defined in a secure channel module
-//included from secure_channel.edl
-
-public int session_setup(
-        [in] setup SessionSetup,
-        [out] SessionSetupResponse);
 
-public int session_close(
-        [in] close SessionClose
-        [out] SessionCloseResponse);
 
-public int session_request(
-        [in] request SessionRequest,
-        [out] SessionRequestResponse;
+Over the [Ledger Enclave - FPC Stub Secure Channel](interfaces.ecc-tlcc-channel.md),
+TLCC_Enclave also offers following RPC functionality to ECC, represented as
+protobuf request and response messages:
 ```
 
+```protobuf
```

Comment:
well, the key part from a spec perspective is the public interface. Internals are more informational and don't have to be here. 

Personally, i think from a design perspective, i see it useful to define also protocol messages and high-level flows to validate that the approach is implementable and also giving a road-map of a (possible) implementation.  The corresponding steps you have to do anyway at some point and writing them up here doesn't really add overhead.  One additional plus is that it also can help to identify early potential some issues with the implementation.  In particularlly nice if has  broader applicability such as on how to define protobufs and hashing/signing (see above discussion leading to refactoring to `ANY`) which also applies to other contexts (e.g. attestation).

In any case, i think in this context it seem worth to me and didn't really added much overhead. But i definitely would not want to force it on everybody. What mainly counts in this API definition "exercise" is that we have complete public APIs with clear semantics, for which as mentioned above you do not need the internals.

Reply:
Let me rephrase: whoever is going to implement this will have to copy-paste what's in here, and keep the two versions in sync.
Our UML diagrams help (i) define and (ii) validate protocols, and (iii) nicely leave some freedom to the developer to organize and modify the low-level implementation. With the level of detail here, we still get (i) and (ii), but we are severely constraining (iii). Again, personal opinion.